spotbugs/src/main/java/edu/umd/cs/findbugs/detect/DumbMethods.java

```diff
@@ -208,13 +208,6 @@ public void sawOpcode(int seen) {
         }
     }
 
-    static int saturatingIncrement(int value) {
-        if (value == Integer.MAX_VALUE) {
-            return Integer.MAX_VALUE;
-        }
-        return value+1;
-    }
-
```

Comment:
I'm not sure why we can remove this logic safely. But the commit which introduces this method e09d217768d636b1eddcb90ca2317d62eacf331d doesn't explain its motivation so clearly... :sob:

Reply:
It is to avoid integer overflow and negative int value instead of a max int.

Reply:
That commit replace an increment operation with a call to incrementSaturating() to avoid integer overflow, but this increment operation is precisely the one that produces the bug. Moreover, since the increment operatation is not needed, there is not risk to overflow.

Reply:
You mean that this increment operation is totally needless, right? Does it mean that existing code misunderstood meaning of 3rd argument of `checkRange()` (e.g. it is 0-based index but we used it as 1-based one)?

Reply:
Yes, at least when checking RANGE_ARRAY_LENGTH and RANGE_ARRAY_OFFSET. Other rules, for instance RANGE_STRING_INDEX works fine (https://github.com/spotbugs/spotbugs/commit/e09d217768d636b1eddcb90ca2317d62eacf331d got rid of the increment operation in that rule)